Spigot-API-Patches/0116-Add-EntityKnockbackByEntityEvent.patch

```diff
@@ -0,0 +1,67 @@
+From c7a07d02e078c5bcd4d155f95d7685a378543fa6 Mon Sep 17 00:00:00 2001
+From: Brokkonaut <hannos17@gmx.de>
+Date: Mon, 18 Jun 2018 15:40:39 +0200
+Subject: [PATCH] Add EntityKnockbackByEntityEvent
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/event/entity/EntityKnockbackByEntityEvent.java b/src/main/java/com/destroystokyo/paper/event/entity/EntityKnockbackByEntityEvent.java
+new file mode 100644
+index 00000000..6194f31d
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/event/entity/EntityKnockbackByEntityEvent.java
+@@ -0,0 +1,52 @@
++package com.destroystokyo.paper.event.entity;
++
++import org.bukkit.entity.Entity;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.entity.EntityEvent;
++import org.bukkit.util.Vector;
++
++/**
++ * Fired when an Entity is knocked back by the hit of another Entity. The acceleration
++ * vector can be modified. If this event is cancelled, the entity is not knocked back.
++ *
++ */
++public class EntityKnockbackByEntityEvent extends EntityEvent implements Cancellable {
++    private static final HandlerList handlers = new HandlerList();
++
++    private final Entity hitBy;
++    private final Vector acceleration;
++    private boolean cancelled = false;
++
++    public EntityKnockbackByEntityEvent(Entity entity, Entity hitBy, Vector acceleration) {
```

Comment:
I don't think a non living entity can be hit by knockback, since the impl is in EntityLiving...

entity should be LivingEntity.

I wish hitBy also was, but this also fires for explosions too, so Entity is valid there.